service/discovery/azure/compute.go

```diff
@@ -70,10 +70,31 @@ func (d *azureComputeDiscovery) List() (list []voc.IsResource, err error) {
 	}
 
 	// Discover virtual machines
+	virtualMachines, _ := d.discoverVirtualMachines()
```

Comment:
Hm. This basically ignores the error in a higher level than before :)

Should be split this into individual discoverers?

Reply:
Why? The error was already handled in the discoverVirtualMachines() function.